test/remove/remove_test.dart

```diff
@@ -22,7 +22,47 @@ void main() {
 
     await d.cacheDir({}).validate();
     await d.appPackagesFile({}).validate();
-    await d.appDir({}).validate();
+    await d.appDir().validate();
+  });
+
+  test('removing a package from dependencies does not affect dev dependencies',
+      () async {
+    await servePackages((builder) {
+      builder.serve('foo', '1.2.3');
+      builder.serve('foo', '1.2.2');
+      builder.serve('bar', '2.0.0');
+    });
+
+    await d.dir(appPath, [
+      YamlDescriptor('pubspec.yaml', '''
+name: myapp
+dependencies: 
+  foo: 1.2.3
+
+dev_dependencies:
+  bar: 2.0.0
+''')
+    ]).create();
+
+    await pubRemove(args: ['foo']);
+
+    await d.cacheDir({'bar': '2.0.0'}).validate();
+    await d.appPackagesFile({'bar': '2.0.0'}).validate();
+
+    final finalPubspec = YamlDescriptor('pubspec.yaml', '''
+name: myapp
+
+dev_dependencies:
+  bar: 2.0.0
+''');
+
+    await d.dir(appPath, [finalPubspec]).validate();
+    final fullPath = p.join(d.sandbox, appPath, 'pubspec.yaml');
+
+    expect(File(fullPath).existsSync(), true);
+
+    final contents = File(fullPath).readAsStringSync();
+    expect(contents, await finalPubspec.read());
```

Comment:
Can we use `not(contains('\ndependencies:'))`? Do we even need this check? as `{dependencies: null}` isn't the same as `{}`.